### DISCUSSION TITLE
Console error after upgrade.

### DISCUSSION THREAD

**Discussion by JorgeSivil** [vercel/next.js#14044] [Help] [answered]
# Bug report

There's a console error after the upgrade to 9.4.4
Doesn't seem to happen with production build.

## Describe the bug

```
Uncaught SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data
    <anonymous> platform.js:14
    NextJS 3
    <anonymous> detect-focus.js:19
    NextJS 3
    <anonymous> supports.js:21
    NextJS 3
    <anonymous> valid-tabindex.js:55
    NextJS 3
    <anonymous> tabindex-value.js:22
    NextJS 3
    <anonymous> focus-relevant.js:19
    NextJS 3
    <anonymous> focusable.js:7
    NextJS 3
    <anonymous> focusable.strict.js:8
    NextJS 3
    <anonymous> focusable.js:42
    NextJS 3
    <anonymous> disabled.js:37
    NextJS 3
    node_modules Overlay.js:29
    NextJS 3
    node_modules index.js:3
    NextJS 3
    node_modules Errors.js:81
    NextJS 3
    node_modules ReactDevOverlay.js:44
    NextJS 3
    node_modules client.js:87
    NextJS 3
    <anonymous> hot-dev-client.js:29
    <anonymous> hot-dev-client.js:375
    NextJS 3
    <anonymous> webpack-hot-middleware-client.js:1
    <anonymous> webpack-hot-middleware-client.js:107
    NextJS 3
    <anonymous> next-dev.js:5
    <anonymous> next-dev.js:121
    NextJS 5
main.js:14:21
    <anonymous> platform.js:14
    NextJS 3
    <anonymous> detect-focus.js:19
    NextJS 3
    <anonymous> supports.js:21
    NextJS 3
    <anonymous> valid-tabindex.js:55
    NextJS 3
    <anonymous> tabindex-value.js:22
    NextJS 3
    <anonymous> focus-relevant.js:19
    NextJS 3
    <anonymous> focusable.js:7
    NextJS 3
    <anonymous> focusable.strict.js:8
    NextJS 3
    <anonymous> focusable.js:42
    NextJS 3
    <anonymous> disabled.js:37
    NextJS 3
    <anonymous> Overlay.js:29
    NextJS 3
    <anonymous> index.js:3
    NextJS 3
    <anonymous> Errors.js:81
    NextJS 3
    <anonymous> ReactDevOverlay.js:44
    NextJS 3
    <anonymous> client.js:87
    NextJS 3
    <anonymous> hot-dev-client.js:29
    <anonymous> hot-dev-client.js:375
    NextJS 3
    <anonymous> webpack-hot-middleware-client.js:1
    <anonymous> webpack-hot-middleware-client.js:107
    NextJS 3
    <anonymous> next-dev.js:5
    <anonymous> next-dev.js:121
    NextJS 

```
## To Reproduce

I had 9.3.4, upgraded NextJS and NextJS/BundleAnalyzer to 9.4.4

Also tested using yarn upgrade --latest

After going to a page, I get that error.

## Expected behavior

No errors should output

## Screenshots

![image](https://user-images.githubusercontent.com/12187148/84323621-84f98d00-ab4d-11ea-912d-0db9ed12eb4d.png)

![image](https://user-images.githubusercontent.com/12187148/84323674-993d8a00-ab4d-11ea-85d0-674a503062af.png)

## System information

- OS: Ubuntu 20.04
- Version of Next.js: 9.4.4
- Version of Node.js: v10.19.0

## Additional context

```
{
  "name": "test-v3",
  "version": "0.0.1",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "dev": "NODE_PATH=. next -p 3000",
    "build": "NODE_PATH=. next build",
    "start": "NODE_PATH=. next start",
    "analyze": "ANALYZE=true NODE_PATH=. next build"
  },
  "dependencies": {
    "@ckeditor/ckeditor5-build-balloon": "^19.0.2",
    "@ckeditor/ckeditor5-font": "^19.0.1",
    "@ckeditor/ckeditor5-react": "^2.1.0",
    "@next/bundle-analyzer": "^9.4.4",
    "babel-eslint": "^10.1.0",
    "babel-plugin-inline-react-svg": "^1.1.1",
    "classnames": "^2.2.6",
    "fetch-retry": "^3.1.0",
    "immer": "^7.0.0",
    "isomorphic-unfetch": "^3.0.0",
    "lodash.clonedeep": "^4.5.0",
    "moment-timezone": "^0.5.31",
    "next": "^9.4.4",
    "next-images": "^1.4.0",
    "node-sass": "^4.14.1",
    "numeral": "^2.0.6",
    "prop-types": "^15.7.2",
    "react": "^16.13.0",
    "react-dom": "^16.13.0",
    "react-elastic-carousel": "^0.6.4",
    "react-imgix": "^9.0.2",
    "react-transition-group": "^4.4.1",
    "sass": "^1.26.8",
    "styled-components": "^5.1.1"
  },
  "devDependencies": {
    "eslint": "^7.2.0",
    "eslint-config-prettier": "^6.11.0",
    "eslint-plugin-prettier": "^3.1.3",
    "eslint-plugin-react": "^7.20.0",
    "prettier": "^2.0.5"
  }
}
```

**Comment by jamesmosier** [comment]
Have you tried removing your `node_modules` and `.next` folder and then reinstalling/rebuilding?

**Reply by JorgeSivil** [reply]
Yes, didn't work

@timneutkens

**Reply by timneutkens** [reply]
```
I had 9.3.4, upgraded NextJS and NextJS/BundleAnalyzer to 9.4.4

Also tested using yarn upgrade --latest

After going to a page, I get that error.


```

If you're @ mentioning make sure your reproduction steps work for anyone instead of "in your application" because this is not enough to look into the issue.

**Reply by JorgeSivil** [reply]
Here are some steps to reproduce https://stackoverflow.com/questions/61857093/json-parse-unexpected-character-at-line-1-column-1-of-the-json-data-when-runing

**Reply by jamesmosier** [reply]
So your issue is due to absolute paths by setting `NODE_PATH`? That's what the Stack Overflow issue is related to.

If so, Next supports [absolute imports](https://nextjs.org/docs/advanced-features/module-path-aliases) now by creating a `jsconfig.json`

Maybe you could provide a GitHub repository showing your specific example.

**Reply by JorgeSivil** [reply]
I switched our config to use jsconfig.json and it works, but the error is still there. I tried to debug it, but couldn't get far.

That library ally.js depends on platform.
It does a require('platform') but then the result of require('platform') is an object that has __esModule: true or something like that.
Then the ally.js code says "if it has __esModule" then return the object, but then later it depends on the object property 'default' which is not there.
That library is not maintained so IDK, but Next depends on it

yarn why v1.22.4
[1/4] Why do we have the module "ally.js"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "ally.js@1.4.1"
info Reasons this module exists
   - "next#@next#react-dev-overlay" depends on it
   - Hoisted from "next#@next#react-dev-overlay#ally.js"
info Disk size without dependencies: "5.24MB"
info Disk size with unique dependencies: "5.3MB"
info Disk size with transitive dependencies: "5.3MB"
info Number of shared dependencies: 2
Done in 0.45s.

**Reply by jamesmosier** [reply]
Unfortunately I won't be of much help without being able to see the project or any code.

**Reply by JorgeSivil** [reply]
I'll try to create the smallest possible repo tomorrow and post back

**Comment by JorgeSivil** [comment, marked as answer]
Error is fixed, the problem seems to have been the NODE_PATH=. variable set in PHPStorm

I recreated the smallest possible thing with only 4 files, and when I ran it in another folder prior to submitting, it worked fine. So I started digging and found out this env in PHPStorm set.

After removing it, the issue was no more present.

Pretty weird tho.

Hate to javascript +1
